src/generators/typescript/presets/JsonBinPackPreset.ts

```diff
@@ -0,0 +1,55 @@
+import { TypeScriptPreset } from '../TypeScriptPreset';
+const alterschema = require('alterschema');
+
+function getInputSchema(originalInput: any) {
+  if(originalInput.$schema !== undefined) {
+    if(originalInput.$schema.includes('http://json-schema.org/draft-04/schema')) {
+      return 'draft4';
+    }
+    if(originalInput.$schema.includes('http://json-schema.org/draft-06/schema')) {
+      return 'draft6';
+    }
+  }
+  return 'draft7';
+}
+
+/**
+ * Preset which adds descriptions
+ *
+ * @implements {TypeScriptPreset}
+ */
+export const TS_JSONBINPACK_PRESET: TypeScriptPreset = {
+  class: {
+    async additionalContent({renderer, content, model}) {
+      renderer.addDependency('const jsonbinpack = require(\'jsonbinpack\');');
```

Comment:
Hmmm... Here you use CommonJS imports. I don't remember but we have a possibility to render dependencies in ESM or CJS, or am I wrong?

Reply:
Ahh, yea, you are right, good catch!

🤔 Need to create some support functions for this then!